server/lib/weather/weather.error.js

```diff
@@ -0,0 +1,9 @@
+class NoWeatherFoundError extends Error {
```

Comment:
NotFoundError form `server/utils/coreErrors` doesn't match the need?

Reply:
The NoWeatherFoundError is used to reply a special message, not a generic message